app/geo/gui/src/ts/twxplore/gui/geo/components/filterSliders/FilterSliders.tsx

```diff
@@ -47,60 +50,66 @@ const FilterSlidersImpl: React.FunctionComponent<{featureType: string}> = ({
-                {attribute}
-              </Typography>
-              <Slider
-                defaultValue={[
-                  attributeProperties.min,
-                  attributeProperties.max,
-                ]}
-                getAriaValueText={valuetext}
-                aria-labelledby="range-slider"
-                step={1}
-                min={attributeProperties.min}
-                max={attributeProperties.max}
-                valueLabelDisplay="auto"
-                disabled={!attributeProperties.max}
-                onChangeCommitted={(event: any, newValue: number | number[]) =>
-                  handleChange(event, newValue, attribute, idx)
-                }
-                name={attribute}
-              />
-            </div>
-          );
-        } else {
-          return <React.Fragment />;
+        const attributeProperties = featureTypeAttributeState[attribute]; //e.g. timestamp:{min,max}, frequency:{min, max}
+        const idx = attributeProperties.filterIndex;
+        switch (featureTypeState) {
+          case MapFeatureTypeState.FILTERS_ADDED: {
+            //if filters have not been set yet. Attach the slider to a filter based on the attribute's unique id
+            dispatch(setFilter(idx, "name", attribute));
+            dispatch(
+              setFilter(
+                idx,
+                "type",
+                getFeatureAttributeByName(attribute).filterType
+              )
+            );
+            dispatch(
+              setFilter(idx, "value", [
+                attributeProperties.min,
+                attributeProperties.max,
+              ])
+            );
+            dispatch(
+              allFiltersSet(
+                FeatureType[featureType as keyof typeof FeatureType]
+              )
+            );
+            return <React.Fragment />;
+          }
+          case MapFeatureTypeState.FILTERS_SET: {
+            return (
+              <div key={attribute}>
+                <Typography id="type" gutterBottom>
+                  {attribute}
+                </Typography>
+                <Slider
+                  defaultValue={[
+                    attributeProperties.min!,
+                    attributeProperties.max!,
+                  ]}
+                  getAriaValueText={valuetext}
+                  aria-labelledby="range-slider"
+                  step={1}
+                  min={attributeProperties.min!}
+                  max={attributeProperties.max!}
+                  valueLabelDisplay="auto"
+                  disabled={!attributeProperties.max!}
+                  onChangeCommitted={(
+                    event: any,
+                    newValue: number | number[]
+                  ) => handleChange(event, newValue, attribute, idx!)}
+                  name={attribute}
+                />
+              </div>
+            );
+          }
+          default: {
+            return <React.Fragment />;
```

Comment:
Should this ever happen? If not, throw an EvalError indicating that there's a missing case.